src/tests/nativeaot/NativeAot.csproj

```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <ItemGroup>
+    <MergedWrapperProjectReference Include="*/**/*.??proj" />
+
+    <!-- Due to the nature of the MultiModule tests, it is unfeasible to convert
+         them to the Merged Wrapper system.
+
+         The main idea of these ones is that we pre-NativeAOT the whole framework,
+         then NativeAOT each assembly separately, and then link them all together
+         at the end using the native linker.
+
+         So, when we try using the Merged Wrapper system, nothing ends up producing
+         the code for xunit.assert, as its assembly is not in the actual framework.
+    -->
+    <MergedWrapperProjectReference Remove="SmokeTests/MultiModule/*.csproj" />
```

Comment:
If we mark the MultiModule tests as "RequiresProcessIsolation", then we shouldn't be getting warnings here. My guess is that I never hooked up the MSBuild logic for the "out of process" tests to exclude them as inputs to ILC. If you want to fix in this this PR, you can. Otherwise, can you file a follow-up issue?

Reply:
I can confirm that adding `<RequiresProcessIsolation>` removes the warnings from this test. So, just to confirm, the follow-up issue is to write the necessary logic to exclude out-of-process tests from being inputs to ILC. Is this accurate?

Reply:
Yep!